src/Microsoft.TestPlatform.Extensions.TrxLogger/ObjectModel/TestPropertyItems.cs

```diff
@@ -0,0 +1,245 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.TestPlatform.Extensions.TrxLogger.ObjectModel
+{
+    using System;
+    using System.Diagnostics;
+    using System.Text;
+
+    using Microsoft.TestPlatform.Extensions.TrxLogger.Utility;
+    using Microsoft.TestPlatform.Extensions.TrxLogger.XML;
+
+    #region TestPropertyItem
+    /// <summary>
+    /// Stores a string which categorizes the Test
+    /// </summary>
+    internal sealed class TestPropertyItem : IXmlTestStore
+    {
+        #region Fields
+        [StoreXmlSimpleField(Location = "Key", DefaultValue = "")]
+        private string key = string.Empty;
+
+        [StoreXmlSimpleField(Location = "Value", DefaultValue = "")]
+        private string value = string.Empty;
+
+        #endregion
+
+        #region Constructors
+        /// <summary>
+        /// Create a new item with the key/value set
+        /// </summary>
+        /// <param name="key">The key.</param>
+        /// <param name="value">The value.</param>
+        public TestPropertyItem(string key, string value)
+        {
+            // Treat null as empty.
+            if (string.IsNullOrEmpty(key))
+                throw new ArgumentNullException();
+
+            if (value == null)
+            {
+                value = String.Empty;
+            }
+
+            this.key = key;
+            this.value = value;
+        }
+
+        #endregion
+
+        #region Properties/Methods
+        /// <summary>
+        /// Gets the Key for this TestProperty
+        /// </summary>
+        public string Key
+        {
+            get
+            {
+                return this.key;
+            }
+        }
+
+        /// <summary>
+        /// Gets the Value for this TestProperty
+        /// </summary>
+        public string Value
+        {
+            get
+            {
+                return this.value;
+            }
+        }
+
+        #endregion
+
+        #region Methods - overrides
+        /// <summary>
+        /// Compare the values of the items
+        /// </summary>
+        /// <param name="other">Value being compared to.</param>
+        /// <returns>True if the values are the same and false otherwise.</returns>
+        public override bool Equals(object other)
+        {
+            TestPropertyItem otherItem = other as TestPropertyItem;
+            if (otherItem == null)
+            {
+                return false;
+            }
+            return String.Equals(this.key, otherItem.key, StringComparison.OrdinalIgnoreCase) && String.Equals(this.value, otherItem.value, StringComparison.Ordinal);
```

Comment:
https://blog.paranoidcoding.com/2019/04/08/string-vs-String-is-not-about-style.html